pinecone/client.go

```diff
@@ -407,11 +512,25 @@ func decodeCollection(resBody io.ReadCloser) (*Collection, error) {
 	return toCollection(&collectionModel), nil
 }
 
-func minOne(x int32) int32 {
-	if x < 1 {
-		return 1
+func ensureHTTPS(inputURL string) (string, error) {
```

Comment:
can we add the option to disable https? (i'm thinking for local dev)

Reply:
The way it's written currently means if you provide a `Host` or `PINECONE_CONTROLLER_HOST`, you're able to use whatever `scheme` as long as `url.Parse()` can read it out. `https` is only applied if there's no scheme, do you think that's sufficient?

For example, you should be able to pass `http://localhost:9999` properly without "https" being added.

Reply:
yup that works. The function name is a bit confusing...